dmpr.py

```diff
@@ -724,195 +686,191 @@ def _map_BW_values(self, node_name, weight_update, dest_name, dest_data):
+                        break
 
     def _add_bandwidthweight_to_dest(self):
         for dest_name, dest_data in self.fib['high_bandwidth'].items():
-            self.fib['high_bandwidth'][dest_name]['weight']=0
+            self.fib['high_bandwidth'][dest_name]['weight'] = 0
             for path_dir, path_name in dest_data['paths'].items():
                 for path_name_fib, path_data_fib in self.fib['path_characteristics'].items():
-                    if path_name==path_name_fib:
-                       self.fib['high_bandwidth'][dest_name]['weight'] = self.fib['high_bandwidth'][dest_name]['weight']+path_data_fib['bandwidth']
-                       break
-
+                    if path_name == path_name_fib:
+                        self.fib['high_bandwidth'][dest_name]['weight'] = self.fib['high_bandwidth'][dest_name][
+                                                                              'weight'] + path_data_fib['bandwidth']
+                        break
 
     def _calc_loss_routingtable(self):
-        self._routing_table['lowest-loss']=list()
+        self._routing_table['lowest-loss'] = list()
         for dest_name, dest_data in self.fib['low_loss'].items():
             for network in dest_data['networks']:
                 loss_entry = dict()
                 for prefix_type, prefix_ip in network.items():
                     loss_entry['proto'] = "v4"
                     ip_pref_len = prefix_ip.split("/")
                     loss_entry['prefix'] = ip_pref_len[0]
-                    loss_entry['prefix-len'] = ip_pref_len[1] 
+                    loss_entry['prefix-len'] = ip_pref_len[1]
                 search_key = '{}>{}'.format(self._conf["id"], dest_data['next-hop'])
                 for path_dir, path_name in dest_data['paths'].items():
-                    if path_dir==search_key:
-                       for fib_path_name, fib_path_data in self.fib['path_characteristics'].items():
-                           if fib_path_name==path_name:
-                              path_found = False
-                              for interface in self._conf['interfaces']:
-                                  path_char = dict()
-                                  path_char = interface['link-characteristics']
-                                  if path_char['loss']==fib_path_data['loss'] and path_char['bandwidth']==fib_path_data['bandwidth']:
-                                     path_found = True
-                                     loss_entry['interface'] = interface['name']
-                                     break
-                              if path_found==True:
-                                 break
-                       break
-                loss_entry['next-hop'] = self.next_hop_ip_addr(loss_entry['proto'], dest_data['next-hop'], loss_entry['interface'])
+                    if path_dir == search_key:
+                        for fib_path_name, fib_path_data in self.fib['path_characteristics'].items():
```

Comment:
I think this is the place where the block should be separated into a own method. But wait until you refactored the whole code and we understand where we wanted to go ...